perf/fateman.jl

```diff
@@ -10,9 +10,9 @@ function fateman1(degree::Int)
     s = TaylorN( [oneH, HomogeneousPolynomial([one(T),one(T),one(T),one(T)],1)], degree )
```

Comment:
The correct answer is "for historical reasons", which somehow do not apply any longer. As it is now, when you compute `s^degree` it only calculates up to `order=degree`, instead of putting lots of zeros (which are homogeneous polynomials of the correct order). I checked the impact in performance of putting `2*degree` there and, with respect to time there is none, but memory increases.

If you want, I can change it, but I don't think it is necessary for this PR.